src/NativeScript/inlineFunctions.js

```diff
@@ -14,6 +14,103 @@ Object.assign(global, {
     },
+
+    // Decorators support    
+    __decorate(decorators, target, key, desc) {
+        var c = arguments.length;
+        var r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc;
+        var d;
+
+        if (typeof global.Reflect === "object" && typeof global.Reflect.decorate === "function") {
+            r = global.Reflect.decorate(decorators, target, key, desc);
+        }
+        else {
+            for (var i = decorators.length - 1; i >= 0; i--) {
+                if (d = decorators[i]) {
+                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
+                }
+            }
+        }
+        return c > 3 && r && Object.defineProperty(target, key, r), r;
+    },
+    __param(paramIndex, decorator) {
+        return function (target, key) {
+            decorator(target, key, paramIndex);
+        }
+    },  
+
+    // Decorators  
+    ObjC(name) {
+        var additionalParams = [];
+        var hasName = (name !== undefined && typeof name === "string");
+        var argsShift = hasName ? 1 : 0;
+        for (var i = argsShift; i < arguments.length; i++) {
+            additionalParams[i - argsShift] = arguments[i];
+        }
+
+        return function (target, propertyKey, descriptor) {
+            if (propertyKey === undefined) { // Class decorator
+                var implementation = {};
+                var baseTarget = Object.getPrototypeOf(target);
+                
+                for (var prop in target.prototype) {
+                    if (target.prototype.hasOwnProperty(prop)) {
+                        implementation[prop] = target.prototype[prop];
+                    }
+                }
+
+                var newName = hasName ? name : target.name;                
+                var extended = baseTarget.extend(implementation, {
+                    name: newName,
+                    protocols: additionalParams.length > 0 ? additionalParams : undefined,
+                    exposedMethods: target.ObjCExposedMethods
+                });
+                extended.name = newName;
+
+                // Add static members
+                for (var prop in target) {
+                    if (target.hasOwnProperty(prop) && prop !== "ObjCExposedMethods") {
+                        extended[prop] = target[prop];
+                    }
+                }
+
+                return extended;
+            }
+
+            // Method Decorator
+            if (!target.constructor.ObjCExposedMethods) {
+                target.constructor.ObjCExposedMethods = {};
+            }
+            if (!target.constructor.ObjCExposedMethods[propertyKey]) {
+                target.constructor.ObjCExposedMethods[propertyKey] = {};
+            }
+            target.constructor.ObjCExposedMethods[propertyKey].returns = additionalParams[0] || interop.types.void;
+
+            if (hasName && name !== propertyKey) {
```

Comment:
Yes, this is exactly the purpose of this. True that it is a bit under the hood. For the JS-friendly name being existent in the proto, I think it is not that big of a problem as you can always make a unique name that will work. I know you can call the method even with the current implementation of the exposed methods. But if you look at the above explicit approach, the code is more or less copy paste with making the same adjustments to every snippet - add a function with the colons removed and call `this` with the colon delimited name and sending the arguments. If you have to do this 100 times, it will not be pretty 😄 But then again, so far, I have never needed to call the exposed method from JS 😉 So I'm ok with both approaaches.